core/src/main/java/de/danoeh/antennapod/core/util/ChapterUtils.java

```diff
@@ -123,6 +135,23 @@ private static CountingInputStream openStream(Playable playable, Context context
         }
     }
 
+    public static List<Chapter> loadChaptersFromUrl(String url) {
+        List<Chapter> chapters;
+        Request request = new Request.Builder().url(url).build();
+        try {
+            Response response = AntennapodHttpClient.getHttpClient().newCall(request).execute();
+            if (response.body() == null) {
+                return null;
+            }
+            chapters = PodcastIndexChapter.parseChapters(response.body().toString());
```

Comment:
Does the body need to be closed? Maybe the `string` method closes it automatically, not sure about `toString`. Getting a StrictMode violation:

```
2022-02-19 14:25:17.429 28336-28364/de.danoeh.antennapod.debug D/StrictMode: StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
        at android.os.StrictMode$AndroidCloseGuardReporter.report(StrictMode.java:1924)
        at dalvik.system.CloseGuard.warnIfOpen(CloseGuard.java:303)
        at java.io.FileOutputStream.finalize(FileOutputStream.java:492)
        at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:291)
        at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:278)
        at java.lang.Daemons$Daemon.run(Daemons.java:139)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.Throwable: Explicit termination method 'close' not called
        at dalvik.system.CloseGuard.openWithCallSite(CloseGuard.java:259)
        at dalvik.system.CloseGuard.open(CloseGuard.java:230)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:253)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:186)
        at okio.Okio.sink(Okio.java:188)
        at okhttp3.internal.io.FileSystem$1.sink(FileSystem.java:49)
        at okhttp3.internal.cache.DiskLruCache$Editor.newSink(DiskLruCache.java:900)
        at okhttp3.Cache$CacheRequestImpl.<init>(Cache.java:442)
        at okhttp3.Cache.put(Cache.java:254)
        at okhttp3.Cache$1.put(Cache.java:149)
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:131)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at de.danoeh.antennapod.core.service.BasicAuthorizationInterceptor.intercept(BasicAuthorizationInterceptor.java:27)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
        at okhttp3.RealCall.execute(RealCall.java:93)
        at de.danoeh.antennapod.core.util.ChapterUtils.loadChaptersFromUrl(ChapterUtils.java:142)
        at de.danoeh.antennapod.core.util.ChapterUtils.loadChapters(ChapterUtils.java:74)
        at de.danoeh.antennapod.core.service.playback.PlaybackServiceTaskManager.lambda$startChapterLoader$2$PlaybackServiceTaskManager(PlaybackServiceTaskManager.java:230)
        at de.danoeh.antennapod.core.service.playback.-$$Lambda$PlaybackServiceTaskManager$K4sCpoV4FY-NLB-eiSWRi0L-X_c.subscribe(Unknown Source:4)
        at io.reactivex.internal.operators.completable.CompletableCreate.subscribeActual(CompletableCreate.java:39)
        at io.reactivex.Completable.subscribe(Completable.java:2302)
        at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
```